tests/HypothesisClient/Serializer/Annotation/Target/Selector/TextQuoteDenormalizerTest.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+namespace tests\eLife\HypothesisClient\Serializer\Annotation\Target\Selector;
+
+use eLife\HypothesisClient\Model\Annotation\Target\Selector\TextQuote;
+use eLife\HypothesisClient\Serializer\Annotation\Target\Selector\TextQuoteDenormalizer;
+use PHPUnit_Framework_TestCase;
+use Symfony\Component\Serializer\Normalizer\DenormalizerInterface;
+
+/**
+ * @covers \eLife\HypothesisClient\Serializer\Annotation\Target\Selector\TextQuoteDenormalizer
+ */
+final class TextQuoteDenormalizerTest extends PHPUnit_Framework_TestCase
+{
+    /** @var TextQuoteDenormalizer */
+    private $normalizer;
+
+    /**
+     * @before
+     */
+    protected function setUpNormalizer()
+    {
+        $this->normalizer = new TextQuoteDenormalizer();
+    }
+
+    /**
+     * @test
+     */
+    public function it_is_a_denormalizer()
+    {
+        $this->assertInstanceOf(DenormalizerInterface::class, $this->normalizer);
+    }
+
+    /**
+     * @test
+     * @dataProvider canDenormalizeProvider
+     */
+    public function it_can_denormalize_text_quotes($data, $format, array $context, bool $expected)
+    {
+        $this->assertSame($expected, $this->normalizer->supportsDenormalization($data, $format, $context));
+    }
+
+    public function canDenormalizeProvider() : array
+    {
+        return [
+            'text-quote' => [[], TextQuote::class, [], true],
+            'non-text-quote' => [[], get_class($this), [], false],
+        ];
+    }
+
+    /**
+     * @test
+     * @dataProvider denormalizeProvider
+     */
+    public function it_will_denormalize_text_quotes(array $json, TextQuote $expected)
+    {
+        $this->assertEquals($expected, $this->normalizer->denormalize($json, TextQuote::class));
+    }
+
+    public function denormalizeProvider() : array
+    {
+        return [
+            'complete' => [
+                [
+                    'exact' => 'exact',
```

Comment:
Not a big deal, but in general in these data-driven tests it's helpful to differentiate values that are used differently, e.g.
````
[
    'exact' => 'e',
    'prefix' => 'p',
    'suffix' => 's',
]
// ...
new TextQuote('e', 'p', 's')
```
so that the relation between values is clear at a glance. Even better would be to have values taken from a real annotation for the `complete` samples.